+transformer/+layer/attention.m

```diff
@@ -63,16 +63,17 @@
 
 % Use the past
 if ~isempty(past)
-    PK = past(:,:,:,1);
-    PV = past(:,:,:,2);
+    PK = permute(past(:,:,:,1,:), [1 2 3 5 4]);
+    PV = permute(past(:,:,:,2,:), [1 2 3 5 4]);
     K = cat(2,PK,K);
     V = cat(2,PV,V);
 end
 
 % Set present. Note that this is done differently from the original
 % implementation which sets the value of present before the previous if
 % statement.
-present = cat(4,K,V);
+present = cat(5,K,V);
+present = permute(present, [1 2 3 5 4]);
```

Comment:
I think what you propose is a better format. My concern was the backwards incompatibility of past/presents associated with the old version, however, since it's a simple fix to update the format we should move ahead with the most appropriate format.

I'll make this change and update the request.